Content.Server/Weapon/Melee/Esword/EswordSystem.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using Content.Shared.Item;
+using Content.Server.Tools.Components;
+using Content.Shared.Interaction;
+using Content.Shared.ActionBlocker;
+using Content.Shared.FixedPoint;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+
+namespace Content.Server.Weapon.Melee.Esword
+{
+    internal class EswordSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<EswordComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<EswordComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<EswordComponent, InteractUsingEvent>(OnInteractUsing);
+
+        }
+
+        private void OnMeleeHit(EntityUid uid, EswordComponent comp, MeleeHitEvent args)
+        {
+            
+            if (comp.Activated == true)
+            {
+                args.BonusDamage.DamageDict = new Dictionary<string, FixedPoint2>()
+                {
+                    {"Slash", FixedPoint2.New(25)},
+                };
+                args.HitSoundOverride = comp.HitSound;
+            }
+            
+        }
+
+        private void OnUseInHand(EntityUid uid, EswordComponent comp, UseInHandEvent args)
+        {
+            if (!Get<ActionBlockerSystem>().CanUse(args.User))
+                return;
+
+            if (comp.Activated)
+            {
+                TurnOff(comp);
+            }
+            else
+            {
+                TurnOn(comp, args.User);
+            }
+        }
+
+        private void TurnOff(EswordComponent comp)
+        {
+            if (!comp.Activated)
+                return;
+
+            if (!EntityManager.TryGetComponent<SpriteComponent?>(comp.Owner, out var sprite) ||
+                !EntityManager.TryGetComponent<SharedItemComponent?>(comp.Owner, out var item))
```

Comment:
Some convenience functions were recently added to make getting components more compact.
Also, after some very recent discussion I think the plan is now to use the convention of having the logical operators on the new line, instead of at the end of the line.

```suggestion
            if (!TryComp(comp.Owner, out SpriteComponent? sprite) 
                || !TryComp(comp.Owner, out SharedItemComponent? item))
```